dat.py

```diff
@@ -18,25 +21,29 @@ def __init__(self, filename):
         self.gene_index = {}
         for i in range(len(self.gene_list)):
             self.gene_index[self.gene_list[i]] = i
+        logger.debug("Got %s genes.", len(self.gene_list))
 
     def open_file(self, filename):
+        logger.debug("Opening %s", filename)
         dab_file = open(filename, 'rb')
 
         # get number of genes
         a = array.array('I')
         a.fromfile(dab_file, 1)
         size = a[0]
+        logger.debug("Expecting %s genes.", size)
 
         # get gene names
         start = 4
         end = 4
         count = 0
-        while count < a[0]:
+        while count < size:
             dab_file.seek(end)
-            if(dab_file.read(2) == '\0\0'):
+            chrs = dab_file.read(2)
+            if(chrs == b'\x00\x00'):
                 dab_file.seek(start)
 
-                gene = dab_file.read(end-start+1)
+                gene = str(dab_file.read(end-start+1))
```

Comment:
@aaronkw : I don't really understand why this has `(end-start+1)`. This serves to grab the null terminator, which seems to be removed in line 48. Happy to remove the +1 and the replace command if that seems reasonable to you. I didn't know if, in some special case that is not true in the dab file that I have, the final character in this case wouldn't be null. Given that we're looking for double nulls above that was hard to imagine.